corehq/apps/export/views.py

```diff
@@ -391,7 +391,14 @@ def prepare_custom_export(self, in_data):
                 )
             export_instances = [self._get_export(self.domain, spec['export_id']) for spec in export_specs]
 
-            self._check_deid_permissions(export_instances)
+            # If any export is de-identified, check that
+            # the requesting domain has access to the deid feature.
+            if not self.permissions.has_deid_view_permissions:
+                for instance in export_instances:
+                    if instance.is_deidentified:
+                        raise ExportAsyncException(
+                            _("You do not have permission to export de-identified exports.")
+                        )
```

Comment:
similarly, I prefer having this abstracted so I don't need to know how deid permissions are checked unless I care to.

Reply:
Agreed. I got into a rhythm of inlining functions for the sake of having fewer class-based methods. Extracted this back into a helper function in 3b3a7650898